ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -869,6 +880,30 @@ function run_upgrade_e2e() {
     echo "e2e test take $(( ($e2e_end_time - $e2e_start_time)/60 )) minutes"
 }
 
+function set_channel(){
+    local x_ver y_ver version="$1"
+    x_ver=$( echo "${version}" | cut -f1 -d. )
+    y_ver=$( echo "${version}" | cut -f2 -d. )
+    ver="${x_ver}.${y_ver}"
+    target_channel="${UPGRADE_CHANNEL}-${ver}"
+    if ! oc adm upgrade channel ${target_channel}; then
+        echo "Fail to change channel to ${target_channel}!"
+        exit 1
+    fi
+    local retry=3
+    while (( retry > 0 )); do
+        versions=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates[]?.version'| xargs)
```

Comment:
we should check `conditionalUpdates` instead of `availableUpdates` here, right? Since availableUpdates existed does not mean end of conditionupdate sync. And i would suggest to move it out of set_channel func and hosted in line623.

Reply:
done

Reply:
and this one miss?

>  i would suggest to move it out of set_channel func and hosted in line623.

1. to align with https://github.com/openshift/release/pull/49540#discussion_r1535044082, make set_channel only do one thing `set channel` in case there will be other scenarios which need set_channel step but not wait for a conditional update check. 
2. it would exit earlier if `[[ " $conditional_updates " =~ " $TARGET_VERSION " ]]`

Reply:
done